libenv/sysinfo.c

```diff
@@ -1058,6 +1058,51 @@ static void SetFlavour(EvalContext *ctx, const char *flavour)
 }
 
 #ifdef __linux__
+
+// NOTE: strings should be canonified before calling
+static void SetFlavour2(
+    EvalContext *const ctx,
+    const char *const id,
+    const char *const major_version)
+{
+    assert(ctx != NULL);
+    assert(id != NULL);
+    assert(major_version != NULL);
+
+    char *flavor;
+    xasprintf(&flavor, "%s_%s", id, major_version);
+    SetFlavour(ctx, flavor);
+    free(flavor);
+}
+
+// Combine version and id into one long canonified string,
+// then define n + 1 hard classes where n is the number of underscores
+// NOTE: strings should be canonified before calling
```

Comment:
Why not use doxygen-style comments? (also applies to the other new function)

Reply:
I don't think this is a big deal for small static functions. If the comment was any longer than this (explaining the parameters etc.) I would use doxygen.

Reply:
Fixed, changed to doxygen.